tests/tracking/test_tracking.py

```diff
@@ -34,6 +32,56 @@ def test_create_experiment():
         tracking.set_tracking_uri(None)
 
 
+def test_get_experiment_by_id():
+
+    service = get_service()
+
+    with pytest.raises(TypeError):
+        service.get_experiment()
+
+    with pytest.raises(Exception):
+        service.get_experiment(-1)
+
+    with pytest.raises(Exception):
+        service.get_experiment(None)
+
+    try:
+        tracking.set_tracking_uri(tempfile.mkdtemp())
```

Comment:
If we're not using `tracking_uri_mock`, could we store the result of `tempfile.mkdtemp` and remove it in our `finally` block to avoid leaking tempdirs? Alternatively, I think it wouldn't be too bad to use `tracking_uri_mock` here - it'd allow us to not worry about manipulating the tracking URI/cleaning up the temp directory, and we'd basically just rewrite the test as:

```
def test_get_experiment_by_name(tracking_uri_mock):

    service = get_service()

    with pytest.raises(TypeError):
        service.get_experiment_by_name()

    with pytest.raises(Exception):
        service.get_experiment_by_name("")

    with pytest.raises(Exception):
        service.get_experiment_by_name(None)

    service = get_service()
    rand_name = "Some random experiment name %d" % random.randint(1, 1e6)
    exp_id = mlflow.create_experiment(rand_name)
    assert exp_id is not None
    experiment = service.get_experiment_by_name(rand_name)
    assert experiment.experiment_id == exp_id
```

I'd prefer to use `tracking_uri_mock`, but also ok to leave this as is & if we clean up the tempdirs.